profiles/utilities.py

```diff
@@ -80,6 +80,21 @@ def remove_none_values(items: dict) -> dict:
     return dict(filter(lambda item: item[1] is not None, items.items()))
 
 
+def dict_contains_none_values(data: dict) -> bool:
+    visited, stack = [], list(data.values())
+    while stack:
+        vertex = stack.pop()
+        if vertex is None:
+            return True
+        if vertex not in visited:
```

Comment:
I think `visited` caters for a dictionary having a reference to itself (a graph), but this should be a tree of values so it wouldn't be necessary?
In general a recursive implementation would be much shorter if it's equivalent:
```
def contains_none_values(data):
    for element in data:
        if element is None:
            return True
       if isinstance(element , dict):
            if contains_none_values(element.values()):
                return True
       if isinstance(element , list):
            if contains_none_values(element):
                return True
return False
```

Reply:
Usually I try to iterate rather than make recursive calls due to pythons recursion limit, however, I think you're right here as the use case for this is for smaller objects rather than large graphs. I have updated the code to use accordingly.